mesonbuild/dependencies/boost.py

```diff
@@ -576,9 +576,6 @@ def detect_lib_dirs(self, root: Path) -> T.List[Path]:
         system_dirs = [x for x in system_dirs if mesonlib.path_is_in_root(x, root)]
         system_dirs = list(mesonlib.OrderedSet(system_dirs))
 
-        if system_dirs:
-            return system_dirs
-
```

Comment:
Removing this effectively makes the above code where `system_dirs` useless, which might break existing setups.

Reply:
I changed it to use both the system_dirs and the hard-coded lib directories